dotnet/src/Planners/Planners.Core/Handlebars/HandlebarsTemplateEngineExtensions.cs

```diff
@@ -143,7 +143,7 @@ internal sealed class HandlebarsTemplateEngineExtensions
             ISKFunction function = kernel.Plugins.GetFunction(functionView.PluginName, functionView.Name);
 
 #pragma warning disable VSTHRD002 // Avoid problematic synchronous waits
-            KernelResult result = kernel.RunAsync(executionContext.Variables, cancellationToken, function).GetAwaiter().GetResult();
+            FunctionResult result = kernel.RunAsync(executionContext.Variables, cancellationToken, function).GetAwaiter().GetResult();
```

Comment:
Can we promote this static to be an async to avoid this diable TSTHRD002?

Reply:
Nope, today it's impossible - https://github.com/microsoft/semantic-kernel/pull/3392#discussion_r1386317690